doc/source/rllib/rllib-offline.rst

```diff
@@ -8,217 +8,662 @@ Working With Offline Data
+``SingleAgentEpisode`` objects with no more than 25 episode objects per Parquet file. Altogether you run 10 evaluation runs, which should
+result in 500 recorded episodes from the expert policy. You will use these data in the next example to train via Offline RL a new policy 
+that should reach a return of 450.0 when playing ``CartPole-v1``.
+
+.. testcode::
+
+    from ray.rllib.algorithms.ppo import PPOConfig
+    from ray.rllib.core import (
+        COMPONENT_LEARNER_GROUP,
+        COMPONENT_LEARNER,
+        COMPONENT_RL_MODULE,
+        DEFAULT_MODULE_ID,
+    )
+    from ray.rllib.core.rl_module import RLModuleSpec
+    
+    # Store recording data under the following path.
+    data_path = "/tmp/docs_rllib_offline_recording"
 
-.. code-block:: bash
+    # Configure the algorithm for recording.
+    config = (
+        PPOConfig()
+        # The environment needs to be specified.
+        .environment(
+            env="CartPole-v1",
+        )
+        # Make sure you use the new API stack.
+        .api_stack(
+            enable_rl_module_and_learner=True,
+            enable_env_runner_and_connector_v2=True,
+        )
+        # Make sure to sample complete episodes because
+        # you want to record RLlib's episode objects.
+        .env_runners(
+            batch_mode="complete_episodes",
+        )
+        # Set up 5 evaluation `EnvRunners` for recording.
+        # Sample 50 episodes in each evaluation rollout.
+        .evaluation(
+            evaluation_num_env_runners=5,
+            evaluation_duration=50,
+        )
+        # Use the checkpointed expert policy from PPO training
+        # above. Note, we have to use the same `model_config` as
+        # the one with which the expert policy was trained, otherwise
+        # the module state cannot be loaded.
+        .rl_module(
+            model_config=DefaultModelConfig(
+                fcnet_hiddens=[32],
+                fcnet_activation="linear",
+                # Share encoder layers between value network
+                # and policy.
+                vf_share_layers=True,
+            ),
+        )
+        # Define the output path and format. In this example you
+        # want to store data directly in RLlib's episode objects.
+        # Each Parquet file should hold no more than 25 episodes. 
+        .offline_data(
+            output=data_path,
+            output_write_episodes=True,
+            output_max_rows_per_file=25,
+        )
+    )
 
-    $ rllib train \
-        --run=PG \
-        --env=CartPole-v1 \
-        --config='{"output": "/tmp/cartpole-out", "output_max_file_size": 5000000}' \
-        --stop='{"timesteps_total": 100000}'
+    # Build the algorithm.
+    algo = config.build()
+    # Load now the PPO-trained `RLModule` to use in recording.
+    algo.restore_from_path(
+        best_checkpoint,
```

Comment:
I think you have to do this here:
```
best_checkpoint / COMPONENT_LEARNER_GROUP / COMPONENT_LEARNER / COMPONENT_RL_MODULE
```
... to make this work ...

Reply:
Really? It worked like this in my example locally ...